app/src/main/java/com/google/android/samples/socialite/ui/camera/CameraViewModel.kt

```diff
@@ -189,19 +191,20 @@ class CameraViewModel @Inject constructor(
             ContextCompat.getMainExecutor(context),
             object : ImageCapture.OnImageSavedCallback {
                 override fun onError(exc: ImageCaptureException) {
-                    val msg = "Photo capture failed."
-                    Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
+                    _imageCaptureState.value = ImageCaptureState.IMAGE_CAPTURE_FAIL
                 }
 
                 override fun onImageSaved(output: ImageCapture.OutputFileResults) {
+                    var state: ImageCaptureState
                     val savedUri = output.savedUri
                     if (savedUri != null) {
+                        state = ImageCaptureState.IMAGE_CAPTURE_SUCCESS
                         sendPhotoMessage(savedUri.toString())
                         onMediaCaptured(Media(savedUri, MediaType.PHOTO))
                     } else {
-                        val msg = "Photo capture failed."
-                        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
+                        state = ImageCaptureState.IMAGE_CAPTURE_FAIL
                     }
+                    _imageCaptureState.value = state
```

Comment:
This could be something like
```kt
  _imageCaptureState.value = if (savedUri != null) {
      sendPhotoMessage(savedUri.toString())
      onMediaCaptured(Media(savedUri, MediaType.PHOTO))
      ImageCaptureState.IMAGE_CAPTURE_SUCCESS
  } else {
      ImageCaptureState.IMAGE_CAPTURE_FAIL
  }
```

At least, make `state` as val and assign success state after `onMediaCaptured(Media(savedUri, MediaType.PHOTO))`.